ContactDetailsApi.Tests/V2/Gateway/DynamoDbGatewayTests.cs

```diff
@@ -70,5 +79,108 @@ public async Task CreateContactDetailsCreatesRecord()
+
+            // Act
+            var response = await _classUnderTest.GetContactDetailsByTargetId(query).ConfigureAwait(false);
+
+            // Assert
+            _logger.VerifyExact(LogLevel.Debug, $"Calling IDynamoDBContext.QueryAsync for targetId {targetId}", Times.Once());
+            response.Should().BeEmpty();
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task GetContactDetailsByTargetIdReturnsExpectedContactDetails(bool hasValidDate)
+        {
+            // Arrange
+            DateTime? validDate = null;
+            if (hasValidDate) validDate = DateTime.UtcNow;
+
+            var entity = _fixture.Build<ContactDetailsEntity>()
+                .With(x => x.RecordValidUntil, validDate)
+                .With(x => x.IsActive, true)
+                .With(x => x.LastModified, validDate)
+                .Create();
+
+            await InsertDataIntoDynamoDB(entity).ConfigureAwait(false);
+
+            var query = new ContactQueryParameter { TargetId = entity.TargetId };
+
+            // Act
+            var result = await _classUnderTest.GetContactDetailsByTargetId(query).ConfigureAwait(false);
+
+            // Assert
+            result.Should().HaveCount(1);
+
+            result.First().Should().BeEquivalentTo(entity, config =>
+            {
+                return config.Excluding(x => x.ContactInformation);
+            });
+
+            _logger.VerifyExact(LogLevel.Debug, $"Calling IDynamoDBContext.QueryAsync for targetId {entity.TargetId}", Times.Once());
+        }
+
+        [Fact]
+        public async Task GetContactDetailsByTargetIdWhenAddressLine1IsEmptyAndContactTypeIsAddressReturnsContentsOfValue()
+        {
+            // Arrange
+            var addressExtended = _fixture.Build<AddressExtended>()
+                .With(x => x.AddressLine1, string.Empty)
+                .Create();
+
+            var contactInformation = _fixture.Build<ContactInformation>()
+                .With(x => x.AddressExtended, addressExtended)
+                .With(x => x.ContactType, ContactDetailsApi.V1.Domain.ContactType.address)
+                .Create();
+
+            var entity = _fixture.Build<ContactDetailsEntity>()
+                .With(x => x.ContactInformation, contactInformation)
+                .With(x => x.RecordValidUntil, DateTime.UtcNow)
+                .With(x => x.IsActive, true)
+                .With(x => x.LastModified, DateTime.UtcNow)
+                .Create();
+
+            await InsertDataIntoDynamoDB(entity).ConfigureAwait(false);
+
+            var query = new ContactQueryParameter { TargetId = entity.TargetId };
+
+            // Act
+            var result = await _classUnderTest.GetContactDetailsByTargetId(query).ConfigureAwait(false);
+
+            // Assert
+            result.First().ContactInformation.AddressExtended.AddressLine1.Should().Be(contactInformation.Value);
```

Comment:
You should do another fluent assertion check to check if result is the same as entity